.circleci/config.yml

```diff
@@ -209,15 +209,23 @@ jobs:
       - <<: *initWorkingDir
       - checkout
       - run: GOOS=linux make build
-      - run: TAG=nightly-${CIRCLE_BRANCH} make docker.all
-      - run: TAG=nightly-${CIRCLE_BRANCH} make test
+      - run: HUB=${DOCKER_USER:-istio} TAG=nightly-${CIRCLE_BRANCH} make docker.all
```

Comment:
I left it as `istio` as that's the default HUB in the Makefile.  IIRC, we had it as `istio` since we originally wanted to push to docker.io/istio but we never got the credentials for that account as the istio project moved away from docker.io, I guess.

Reply:
re: the password... in the circleci project settings for istio-cni, they recommend using environment variables that are only able to be seen by circleci admins of the project.  It's still not great but it's at least not on public github.

Reply:
Perhaps its your PR heading comment that is confusing,   but this change set doesn't include the istiocni value in the change set at all.   So I think what you meant is this change allows pushing to an account that is passed in via environmental value.  If that is what you intended then the change set looks good.   If that is not what you intended then I don't see how this gets pushed to istiocni

Reply:
Yeah originally, I had changed the Makefile's default HUB value to istiocni but that causes problems due to assumptions in the circleci test code that I don't think are worth fixing.  So, yes, the circleci `DOCKER_USER=istiocni` is what makes this push to `istiocni` since we have credentials for that account.  Since DOCKER_USER was never set for the istio/cni circleci env this code never ran prior and always would use the Makefile's HUB value regardless of the DOCKER_USER setting which is not correct.